### ISSUE TITLE
Styleguide: Exploring the Storybook alternative

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#3388]
We currently don't give a lot of attention to our styleguide, yet it's a powerful tool to document, develop and prototype quickly.

Last time I looked, the main differences with the system we have today (styleguidist) were:
- Styleguidist uses markdown files, storybook is pure js (you write "stories"). It's easier to write custom pages with Styleguidist
- Storybook seems to have a more mature ecosystem
- Storybook "stories" look like tests, and actually act as such (ie. if you change a component props without updating the story, the CI will fail)

The expected benefits if we were to change tomorrow would mostly be:
- A better UI for our styleguide
- Preventing regressions: on the current styleguide, if you break an example by changing a component, you won't be notified. If you do so with Storybook, you'll have to fix it before you can merge your work.

## MVP
To demonstrate the change, I suggest that we create a PR with storybook installed that migrates 3 components. This PR should document how much time it takes to migrate a component and what are the eventual drawbacks & benefits.

**Comment by flickz** [comment]
> Styleguidist uses markdown files, storybook is pure js (you write "stories"). It's easier to write custom pages with Styleguidist

Storybook has a markdown feature, called [`MDX`](https://storybook.js.org/docs/react/writing-docs/mdx), I'm not sure exactly how it works but I think it's worth exploring to make the migration faster. 

I love the idea of having a better UI for the styleguide, one of my favorites is [Atlassian](https://atlassian.design/)'s design system docs.

**Comment by Betree** [comment]
@flickz thanks for sharing, MDX sounds super interesting to migrate more easily and to make sure we preserve the static documentations pages (ie. typography, grid, etc).

**Comment by Betree** [comment]
Our styleguide is currently broken - though https://styleguide.opencollective.com still hosts an old version that used to work. I tried solving the dependency conflict (on `ast-types`) without success. Also tried to roll it back to `11.0.0`, there again with no success. 

The errors are switching between these two lines (depending on the various fixes I tried):
> Error: did not recognize object of type "ChainExpression"
> RangeError: Maximum call stack size exceeded

Given that Storybook is more maintained, it would be great to migrate if there's an easy path. Otherwise, without proper prioritization, I'm afraid we will have to kill styleguidist :crying_cat_face:
